### PR TITLE
Added kasai_algorithm

### PR DESCRIPTION
**Pull request by jujusudhur**
**Fixes issue:** #2999 

**Changes:**
added kasai Algorithm 
Path : code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp

### CODE REVIEW

File: code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* 
+ *
+ * Kasai Algorithm 
+ * 
+ */
+
+vector < int > kasaiAlgorithm(string s , vector < int > suffix_array)
```
Comment (arnavb) [inline code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp]: No need for the spaces with the templates (C++11).

File: code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    int n = s.size() , m = 0;
```
Comment (arnavb) [inline code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp]: The variable n is unnecessary.

File: code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    vector < int > LCP(n , 0);
+    vector < int > rank(n , 0);
+
+    for (int i = 0 ; i < n ; i ++) rank[suffix_array[i]] = i;
+
+    for (int i = 0 ; i < n ; i ++ , m ? m -- : 0)
+    {
+        if (rank[i] == n - 1) 
+            {
```
Comment (arnavb) [inline code/string-algorithms/kasai_algorithm/kasai_algorithm.cpp]: Fix formatting.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**